<launch>
    <arg name="cam_raw_topic" value="/rgb/image_rect_color" />
    <arg name="cam_raw_info_topic" value="/rgb/camera_info" />
    <arg name="cam_out_topic" value="/rgb/proc/image_rect_color" />
    <arg name="cam_out_info_topic" value="/rgb/proc/camera_info" />

    <!-- Downsample and crop image -->
    <node pkg="nodelet" type="nodelet" name="image_crop_decimate"
        args="standalone image_proc/crop_decimate standalone_nodelet">

        <remap from="camera/image_raw" to="$(arg cam_raw_topic)" />
        <remap from="camera/camera_info" to="$(arg cam_raw_info_topic)" />

        <remap from="camera_out/image_raw" to="$(arg cam_out_topic)" />
        <remap from="camera_out/camera_info" to="$(arg cam_out_info_topic)" />

        <param name="decimation_x" type="int" value="3" />
        <param name="decimation_y" type="int" value="3" />
        <param name="x_offset" type="int" value="50" />
        <param name="y_offset" type="int" value="50" />
        <param name="width" type="int" value="1820" />
        <param name="height" type="int" value="980" />
    </node>

    <arg name="config_path" default="$(find hl_orbslam3_wrapper)/cfg/vinsmono_spot.yaml" />
    <arg name="vins_path" default="$(find feature_tracker)/../config/../" />

    <node name="feature_tracker" pkg="feature_tracker" type="feature_tracker" output="screen">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="vins_folder" type="string" value="$(arg vins_path)" />
    </node>

    <node name="vins_estimator" pkg="vins_estimator" type="vins_estimator" output="screen">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="vins_folder" type="string" value="$(arg vins_path)" />
    </node>

    <node name="pose_graph" pkg="pose_graph" type="pose_graph" output="screen">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="visualization_shift_x" type="int" value="0" />
        <param name="visualization_shift_y" type="int" value="0" />
        <param name="skip_cnt" type="int" value="0" />
        <param name="skip_dis" type="double" value="0" />
    </node>

</launch>
